scripts/dts/extract/globals.py

```diff
@@ -287,21 +287,21 @@ def add_prop_aliases(node_path,
 def get_binding(node_path):
     compat = get_compat(node_path)
 
-    # For just look for the binding in the main dict
-    # if we find it here, return it, otherwise it best
-    # be in the bus specific dict
-    if compat in bindings:
-        return bindings[compat]
-
+    # First look for a bus-specific binding. If not found, look in the
+    # main dict.
     parent_path = get_parent_path(node_path)
     parent_compat = get_compat(parent_path)
 
-    parent_binding = bindings[parent_compat]
+    try:
+        parent_binding = bindings[parent_compat]
+        bus = parent_binding['child']['bus']
+        binding = bus_bindings[bus][compat]
+        return binding
+    except (KeyError, TypeError):
+        pass
 
-    bus = parent_binding['child']['bus']
-    binding = bus_bindings[bus][compat]
-
-    return binding
+    if compat in bindings:
+        return bindings[compat]
```

Comment:
Maybe this could just be `return bindings[compat]`. That way the code will fail earlier if it's assumed that `get_binding()` always returns a binding (and it makes it explicit that that's the assumption).

Reply:
Looks like some code calls `get_binding()` and checks for exceptions to see if no binding was found. For that code to keep working, it'll need to be `return bindings[compat]`.

Might be cleaner to have `get_binding()` explicitly `return None` and check for that in the code instead of catching exceptions. That's an unrelated refactoring though.